WordPress/Classes/ViewRelated/Menus/Views/MenuItemEditingHeaderView.m

```diff
@@ -0,0 +1,212 @@
+#import "MenuItemEditingHeaderView.h"
+#import "WPStyleGuide.h"
+#import "MenuItem.h"
+#import "MenuItem+ViewDesign.h"
+
+@interface MenuItemEditingHeaderView () <UITextFieldDelegate>
+
+@property (nonatomic, strong, readonly) UIStackView *stackView;
+@property (nonatomic, strong, readonly) NSLayoutConstraint *stackViewTopConstraint;
+@property (nonatomic, strong, readonly) UIView *textFieldContainerView;
+@property (nonatomic, strong, readonly) UIImageView *iconView;
+
+@end
+
+@implementation MenuItemEditingHeaderView
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)awakeFromNib
+{
+    [super awakeFromNib];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(deviceOrientationDidChangeNotification) name:UIDeviceOrientationDidChangeNotification object:nil];
+    
+    self.translatesAutoresizingMaskIntoConstraints = NO;
+    self.backgroundColor = [UIColor clearColor];
+
+    [self setupStackView];
+    [self setupIconView];
+    [self setupTextField];
```

Comment:
For this one, a couple of reasons. Storyboards/nibs end up being a bit more problematic with UIStackViews that are not straightforward setups, at least for this version of Xcode. Additionally, these views are setup based on the shared design constants which is easier to keep in-check with a code setup.

That said, I generally I tend to avoid doing a lot of setup in both a nib and code and either laying mostly into one or the other. Let me know if you have any suggestions otherwise.